state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2092,10 +2072,28 @@ impl<T: Config<I>, I: 'static> DepositApi<T::TargetChain> for Pallet<T, I> {
 		(ChannelId, ForeignChainAddress, <T::TargetChain as Chain>::ChainBlockNumber, Self::Amount),
 		DispatchError,
 	> {
+		let swap_limits = T::SwapLimitsProvider::get_swap_request_limits();
 		if let Some(params) = &refund_params {
 			ensure!(
-				params.retry_duration <= MaxSwapRetryDurationBlocks::<T, I>::get(),
-				DispatchError::Other("Retry duration too long")
+				params.retry_duration <= swap_limits.max_swap_retry_duration_blocks,
+				DispatchError::from(Error::<T, I>::SwapRetryDurationTooLong)
+			);
+		}
+
+		if let Some(params) = &dca_params {
+			ensure!(
+				params.number_of_chunks > 0 && params.chunk_interval >= SWAP_DELAY_BLOCKS,
```

Comment:
To avoid checking SWAP_DELAY_BLOCKS here (and to keep this swapping-specific configuration isolated to the swapping pallet), we could just add the SWAP_DELAY_BLOCK to the chunk interval when we schedule the next chunk? Arguably this is more consistent with how we execute the first block: when we submit an initial swap, it is always scheduled for execution after three blocks. 

This also avoid having to communicate to users that there's a minimum chunk interval (ie. 0 just means reschedule the next chunk as soon as the first on is complete). Thoughts? (cc @msgmaxim )

Reply:
This is more of a product question, but personally I prefer keeping it the way it is. I think saying that there is a mimimum delay of 2 blocks is at least as (if not more) intuitive than having a hidden 2 block delay. 

> Arguably this is more consistent with how we execute the first block

I think we there are reasonable arbuments both ways. I think what's important is that we consistently notify LPs of a swap at least 2 blocks in advance. From the code's perspective I don't think we are too inconsistent: `schedule_swap` takes a block delay in all cases. It is just that for the first chunk there is a different delay since "interval" doesn't apply, and I think this is justified as the first chunk is literarly an edge case. Not sure if subtracting 2 from `chunk_interval` just so it can be added back at a later point makes things better. 

> to keep this swapping-specific configuration isolated to the swapping pallet

This is a good point, but I think the problem lies somewhere deeper. The two pallets are already tightly coupled: ingres-gress knows about channels which know about swaps. We should either fix this (not even sure if it is possible/desirable), or accept that there is strong coupling and not worry about workarounds to hide the symptoms.